numpyro/distributions/distribution.py

```diff
@@ -71,7 +71,23 @@ def validation_enabled(is_validate=True):
         enable_validation(distribution_validation_status)
 
 
-class Distribution(object):
+# TODO Remove import guard once funsor is a required dependency.
```

Comment:
Could you refactor to use the updated `COERCIONS` pattern as in https://github.com/pyro-ppl/pyro/pull/2641 ? I'm hoping we can refactor `contrib.epidemiology` to use the same `COERCIONS` mechanism, thus keeping Numpyro aligned with Pyro will make it easier to eventually port `contrib.epidemiology` to Numpyro.

Reply:
Sure, I didn't notice that you put a new pattern in Pyro.